garrysmod/lua/includes/extensions/string.lua

```diff
@@ -289,12 +289,10 @@ local meta = getmetatable( "" )
 
 function meta:__index( key )
 	local val = string[ key ]
-	if ( val ) then
+	if ( val ~= nil ) then

```

Comment:
Is it more accurate when `key` is nil to raise error ? This way `("test")[nil]` will do nothing.

Reply:
All this change does is allow
```Lua
string.foo = false
print(string.foo)
```
to not error.